strings/c-or-cpp/Permutation-of-String.c

```diff
@@ -55,11 +55,13 @@ void swap(char *a,char *b)
     *b=temp;
 	//swapping done
 }
-// Example:
-//Input: abc
-//Output: 1(abc)
-//        2(acb)
-//        3(bac)
-//        4(bca)
-//        5(cba)
-//        6(cab)
+/* Example:
+  Input: abc
+  Output: 1(abc)
+          2(acb)
+          3(bac)
+          4(bca)
+          5(cba)
+          6(cab)
+Time Complexity: O(n*n!)
+*/
```

Comment:
did i need to add new line at the end

Reply:
Although it is recommended to end the file with newline character, It is no hard and fast rule. 

The reason behind this lies in the POSSIX definition of a line where by convention, termination of line is defined by a newline (or more formally end of line <EOL> ) character, so your last line would not be considered as a line till the time you don't add newline at the end of it.

Although this should not be a problem till the time you are not using unix tools ( like concatenation, which might give undesired results when used with files with no newline at the end) , It is recommended to follow the convention and add a newline at the end of the file.